PINCache/PINDiskCache.m

```diff
@@ -1107,10 +1107,16 @@ - (void)unlock
 @end
 
 @implementation PINBackgroundTask
+
++ (BOOL)isAppExtension
+{
+    return [[[NSBundle mainBundle] executablePath] containsString:@".appex/"];
```

Comment:
Can you write a unit test for this? I think this is doable if you separated this into 2 methods, once of which just took a string and return a BOOL, and `isAppExtension` could  all that one with `[[NSBundle mainBundle] executablePath]`